bin/citrea/tests/test_helpers/mod.rs

```diff
@@ -156,8 +162,96 @@ pub fn tempdir_with_children(children: &[&str]) -> TempDir {
     let db_dir = tempfile::tempdir().expect("Could not create temporary directory for test");
     for child in children {
         let p = db_dir.path().join(child);
-        std::fs::create_dir(p).unwrap();
+        if !std::path::Path::new(&p).exists() {
+            std::fs::create_dir(p).unwrap();
+        }
     }
 
     db_dir
 }
+
+pub async fn wait_for_l2_batch(sequencer_client: &TestClient, num: u64, timeout: Option<Duration>) {
+    let start = SystemTime::now();
+    let timeout = timeout.unwrap_or(Duration::from_secs(30)); // Default 30 seconds timeout
+    loop {
+        debug!("Waiting for soft batch {}", num);
+        let latest_block = sequencer_client
+            .eth_get_block_by_number_with_detail(Some(BlockNumberOrTag::Latest))
+            .await;
+        if latest_block.number >= Some(num.into()) {
+            break;
+        }
+
+        let now = SystemTime::now();
+        if start + timeout <= now {
+            panic!("Timeout");
+        }
+
+        sleep(Duration::from_secs(1)).await;
+    }
+}
+
+pub async fn wait_for_prover_l1_height(
+    prover_client: &TestClient,
+    num: u64,
+    timeout: Option<Duration>,
+) {
+    let start = SystemTime::now();
+    let timeout = timeout.unwrap_or(Duration::from_secs(30)); // Default 30 seconds timeout
+    loop {
+        debug!("Waiting for prover height {}", num);
+        let latest_block = prover_client.prover_get_last_scanned_l1_height().await;
+        if latest_block >= num {
+            break;
+        }
+
+        let now = SystemTime::now();
+        if start + timeout <= now {
+            panic!("Timeout");
+        }
+
+        sleep(Duration::from_secs(1)).await;
+    }
+}
+
+pub async fn wait_for_l1_block(da_service: &MockDaService, num: u64, timeout: Option<Duration>) {
+    let start = SystemTime::now();
+    let timeout = timeout.unwrap_or(Duration::from_secs(30)); // Default 30 seconds timeout
+    loop {
+        debug!("Waiting for L1 block height {}", num);
+        let da_block = da_service.get_height().await;
+        if da_block >= num {
+            break;
+        }
+
+        let now = SystemTime::now();
+        if start + timeout <= now {
+            panic!("Timeout");
+        }
+
+        sleep(Duration::from_secs(1)).await;
+    }
+}
+
+pub async fn wait_for_commitment(
```

Comment:
i think we should indicate this checks commitments on postgres, the name should reflect that